Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/FieldsAndConstructor.tt

```diff
@@ -31,6 +39,30 @@
 <#
     if (_model.LambdaMethod.UsingDependencyInjection)
     {
+
+        if (_model.LambdaMethod.UsingHostBuilderForDependencyInjection)
+        {
+#>
+            var startup = new <#= _model.StartupType.FullName #>();
+            var hostBuilder = startup.ConfigureHostBuilder();
+
+            // By default, Lambda function class is added to the service container using the singleton lifetime
+            // To use a different lifetime, specify the lifetime in Startup.ConfigureServices(IServiceCollection) method.
+            hostBuilder.Services.AddSingleton<<#= _model.LambdaMethod.ContainingType.Name #>>();
+            hostBuilder.Services.AddSingleton<<#= _model.SerializerInfo.SerializerName #>>();
+
+            serviceProvider = hostBuilder.Services.BuildServiceProvider();
+
+            host = hostBuilder.Build();
```

Comment:
IHostApplicationBuilder doesnt technically have a `build` function in its interface so we are assuming it exists.  not sure if theres a better way